modules/optimization/doc/content/modules/optimization/getting_started/InvOptTheory.md

```diff
@@ -0,0 +1,376 @@
+
+We have two options for dealing with the inverse of the Jacobian.  In the first, we could perform one linear solve per parameter, $p_i$, being fit by solving $\left[\left(\frac{\partial g}{\partial \mathbf{u}}\right)^{-1}\frac{\partial g}{\partial p_i}\right]$.  This algorithm scales with the number of parameters which makes it computationally expensive for tens of parameters.
+
+The alternative that we will use is the adjoint method which scales independently to the number of optimization parameters.   The adjoint method requires one additional solve of the same complexity as the original forward problem.  The adjoint equation given in [eq:adjoint] is found by setting the adjoint variable, $\lambda$, equal to the first two terms of [eq:objectiveGrad2] and then rearranging terms to give
+\begin{equation} \label{eq:AdjEqn}
+	\begin{aligned}
+	&\lambda^\top= \frac{\partial J}{\partial \mathbf{u}}\left(\frac{\partial g}{\partial \mathbf{u}}\right)^{-1}  \\
+	&\rightarrow \lambda^\top \left(\frac{\partial g}{\partial \mathbf{u}}\right)= -\frac{\partial J}{\partial \mathbf{u}}\left[\left(\frac{\partial g}{\partial \mathbf{u}}\right)^{-1} \left(\frac{\partial g}{\partial \mathbf{u}}\right)\right] \\
+	&\rightarrow \lambda^\top \left(\frac{\partial g}{\partial \mathbf{u}}\right)= -\frac{\partial J}{\partial \mathbf{u}} \\
+	&\rightarrow \left[\lambda^\top \left(\frac{\partial g}{\partial \mathbf{u}}\right)\right]^\top= -\left[\frac{\partial J}{\partial \mathbf{u}}\right]^\top \\
+	&\rightarrow \left(\frac{\partial g}{\partial \mathbf{u}}\right)^\top \lambda = -\left[\frac{\partial J}{\partial \mathbf{u}}\right]^\top \\
+\end{aligned}
+\end{equation}
+where each step of the derivation has been included as a reminder of how [eq:adjoint] is obtained.  The next [section](#sec:adjoint) uses an alternative approach to determine the adjoint equation based on Lagrange multipliers.
+
+[comment1]: <> (% ----------------------------------------------------------------------------------------------------------%)
+
+## Adjoint Problem for Steady State Heat Conduction id=sec:adjoint
+
+In this section, we are interested in solving the following PDE-constrained optimization problem from [eq:optimization] for steady state heat conduction:
+\begin{equation} \label{eqn:optimization_problem}
+	\begin{aligned}
+	& \min_{\mathbf{p}}
+	& & J\left(T,\mathbf{p}\right) = \frac{1}{2} \sum_{i=1}^{N} \left( T_i - \bar{T}_i \right)^2 , \\
```

Comment:
I guess adding square root of J won't change the results of variational derivation. Possibly can add a sentence to comment on this.